### ISSUE TITLE
Example use with Gin Framework

### ISSUE THREAD

**Issue by batara666** [jptosso/coraza-waf#56]
Your's such great & useful software

**Comment by jptosso** [comment]
Gin framework seems interesting, I'm going to add it to the list of possible modules and middlewares

Thank you!

**Comment by jptosso** [comment]
I might create a working plugin in the future but this will work for phases 1, 2 and 5. Phases 3 and 4 requires a responseWriter overwrite.

```
func Coraza() gin.HandlerFunc {
        waf := engine.NewWaf()
        parser := seclang.NewParser(waf)
        parser.FromString(`secrule.....`)
	return func(c *gin.Context) {
		tx := waf.NewTransaction()
                 tx.ProcessRequest(c.Request)
                 if tx.Interruption != nil {...}
		c.Next()
		// in order to parse the response, you must overwrite the ResponseWriter https://github.com/gin-gonic/gin/blob/caf2802593277033683c4e8cb5f22c81cc35eae8/response_writer.go#L20
	}
}

func main() {
	r := gin.New()
	r.Use(Coraza())

	r.GET("/test", func(c *gin.Context) {
// your code
	})

	// Listen and serve on 0.0.0.0:8080
	r.Run(":8080")
}
```

**Comment by jptosso** [comment]
Now there is a gin implementation: https://github.com/jptosso/coraza-gin

**Comment by wprimadi** [comment]
> Now there is a gin implementation: https://github.com/jptosso/coraza-gin

I hope this message finds you well. I am reaching out to kindly request support for Coraza version 3 compatibility in the coraza-gin middleware library.

I've noticed that the current version of coraza-gin supports Coraza version 2, but as Coraza version 3 has been released with notable enhancements, I believe that updating the middleware to support Coraza v3 would greatly benefit the community.

Coraza version 3 introduces significant changes, I understand that this might require adjustments in the middleware's codebase. Is there a roadmap or plan in place to bring coraza-gin up to date with Coraza version 3?
